Hypermedia.AspNetCore.Siren/ServiceCollectionExtensions.cs

```diff
@@ -5,22 +5,35 @@
     using Actions.Fields.Type;
     using Actions.Fields.Validation;
     using Endpoints;
-    using Entities.Builder;
+    using Entities;
+    using Environments;
+    using Microsoft.AspNetCore.Http;
     using Microsoft.Extensions.DependencyInjection;
+    using Resources;
 
     public static class ServiceCollectionExtensions
     {
         public static IMvcBuilder AddHypermediaSiren(this IMvcBuilder mvcBuilder)
         {
             var services = mvcBuilder.Services;
 
-            services.AddTransient<EntityBuilder>();
             services.AddSingleton<IAccessValidator, AccessValidator>();
             services.AddSingleton<IActionDescriptorResolver, ActionDescriptorResolver>();
             services.AddSingleton<ICallCollector, ExpressionCallCollector>();
             services.AddSingleton<IEndpointDescriptorProvider, EndpointDescriptorProvider>();
             services.AddSingleton<IHrefFactory, HrefFactory>();
             services.AddSingleton<IFieldsFactory, FieldsFactory>();
+            services.AddSingleton(typeof(IScopedBuildApplier<,>), typeof(ScopedBuildApplier<,>));
+
+            services.AddAsyncBuildSteps();
+
+            services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+            services.AddSingleton(s => s.GetService<IHttpContextAccessor>().HttpContext.User);
+
+            services.AddTransient<IEntityBuilder, EntityBuilder>();
+            services.AddTransient(typeof(IStorage<>), typeof(Storage<>));
+            services.AddTransient(typeof(IAsyncBuildingEnvironment<,>), typeof(AsyncBuildingEnvironment<,>));
+            services.AddTransient<IResourceBuilder, ResourceBuilder>();
```

Comment:
We could extract these into a separate function similar to "AddAsyncBuildSteps".

Reply:
I'm planning on doing this in a separate branch, actually moving code in another project where all of this will be encapsulated.